admin-console/src/main/webapp/src/components/SystemBehaviorGroups/BehaviorGroupTable.tsx

```diff
@@ -0,0 +1,182 @@
+
+    const onClose = () => {
+        setShowModal(false);
+        getBehaviorGroups.query;
+    };
+
+    const onDeleteClose = () => {
+        setShowDeleteModal(false);
+        getBehaviorGroups.query;
+    };
+
+    if (getBehaviorGroups.loading) {
+        return <Spinner />;
+    }
+
+    if (getBehaviorGroups.payload?.status !== 200) {
+        return <span>Error while loading sysem behavior groups: {getBehaviorGroups.errorObject.toString()}</span>;
+    }
+
+    return (
+        <React.Fragment>
+            <PageSection>
+                <Title headingLevel='h1'>
+                    <Breadcrumb>
+                        <BreadcrumbItem target='#' >{ (getBundles.loading || getBundles.payload?.status !== 200)
+                            ? <Spinner /> : getBundles.payload.value.displayName }
+                        </BreadcrumbItem>
+                        <BreadcrumbItem target='#'> System Behavior Groups </BreadcrumbItem>
+                    </Breadcrumb>
+                </Title>
+                <TableComposable aria-label="System behavior groups table">
+                    <Thead>
+                        <Toolbar>
+                            <ToolbarContent>
+                                <ToolbarItem>
+                                    <Button variant='primary' type='button' onClick={ createBehaviorGroup }> Create new group </Button>
+                                    <CreateEditBehaviorGroupModal
+                                        isEdit={ isEdit }
+                                        isOpen={ isOpen }
+                                        initialSystemBehaviorGroup={ systemBehaviorGroup }
+                                        showModal={ showModal }
+                                        onClose={ onClose }
+                                        onSubmit={ isEdit ? handleUpdate : handleSubmit }
+                                        isLoading={ false }
+                                    />
+                                    <DeleteBehaviorGroupModal
+                                        onDelete={ handleDelete }
+                                        bundleName={ bundle?.displayName }
+                                        systemBehaviorGroupName={ systemBehaviorGroup.displayName }
+                                        isOpen={ showDeleteModal }
+                                        onClose={ onDeleteClose }
+                                    />
+                                </ToolbarItem>
+                            </ToolbarContent>
+                        </Toolbar>
+                        <Tr>
+                            {columns.map((column, columnIndex) => (
+                                <Th key={ columnIndex }>{column}</Th>
+                            ))}
+                        </Tr>
+                    </Thead>
+                    <Tbody>
+                        { getBehaviorGroups.payload.value.map(b =>
```

Comment:
Filtering the bg's I tried add it here, but it broke everything. I tried similar to what you suggested but I think my placement is wrong. `getBehaviorGroups.payload.value.filter( bg => bg.bundleId === bundleId (utilizing the useParams above)`

Reply:
It could be easier if you split the data from the ui components. You can abstract the logic of the filtering in a separate component and just passing params to  a `BehaviorGroupTable` component.

Try to not mix the UI and the logic